### PR TITLE
build: update to angular v12.0.0-rc.0

### PR DESCRIPTION
**Pull request by devversion**
Updates to Angular v12.0.0-rc.0

This unblocks #22351 with https://github.com/angular/angular/commit/ab257b370127dce70bd3ee7ad6d64d3a9ad5ae95 being available.

### CODE REVIEW

**Review by jelbourn — APPROVED** [review]
lgtm

File: package.json [hunk 1]
```diff
@@ -74,12 +74,12 @@
     "@angular-devkit/build-optimizer": "^0.1200.0-next.9",
```
Comment (crisbeto) [inline package.json:74]: Should the CLI dependencies be updated as well?

Comment (devversion) [reply package.json:74]: Yeah, will update. CLI rc.0 wasn't available when I submitted this PR:

Comment (devversion) [reply package.json:74]: @crisbeto done.

**Review by crisbeto — APPROVED** [review]
LGTM

File: package.json (hunk 1 above)
Comment (Splaktar) [reply package.json:74]: @devversion it looks like you only bumped 2 of the 3 CLI deps to rc.0?

Comment (devversion) [reply package.json:74]: You're right. I'll send a follow-up.

**Pull request merged**